superduperdb/components/component.py

```diff
@@ -67,25 +68,29 @@ class Component(Serializable, Leaf):
     artifacts: dc.InitVar[t.Optional[t.Dict]] = None
     changed: t.ClassVar[set] = set([])
 
+    def __post_init__(self, db, artifacts):
+        super().__post_init__(db)
+
+        self.artifacts = artifacts
+        self.version: t.Optional[int] = None
+        self._db = None
```

Comment:
Remove this? We have set the db in `super().__post_init__(db)`

Reply:
good catch!